Dockerfile

```diff
@@ -2,10 +2,11 @@ FROM registry.ci.openshift.org/origin/4.12:artifacts as artifacts
 
 FROM quay.io/fedora/fedora-coreos:stable
 ARG FEDORA_COREOS_VERSION=412.37.1
+ARG ARCHITECTURE=x86_64
```

Comment:
Hello,

Dockerfiles should not need external CLIs to pass platform-specific arguments. Building on a given platform should produce a container image for that platform by default, i.e.,: 

```
podman build .
```

It has to produce an arm64 binary in arm64 hosts, an amd64 in amd64 build hosts, and so on.

A few default arguments are available with specific versions of Docker upstream, but this is yet to be available in most of the tools we use in our CI systems. Therefore, such a PR would compromise a build for a platform different than x86_64 (currently wip for okd/fcos).


The ideal way is still the workaround available in the master branch: https://github.com/openshift/okd-machine-os/blob/master/Dockerfile#L25-L26. It will work fine on any platform you build the image on.

Reply:
Should we cherry-pick https://github.com/openshift/okd-machine-os/pull/462 ?

Reply:
Ah, so OKD machine OS image would be built with whatever arch `registry.ci.openshift.org/origin/4.x:artifacts` is built with and its just a matter of cherrypicking changes from master?

@dsevost does that sound good to you?

Reply:
yes, sounds reasonable, 
anyway if you going to build image for arm64 on x86 this can be useful
```
echo "/tmp/rpms/$([ -z "$ARHITECTURE" ] && { [ -d /tmp/rpms/$(uname -m) ] && echo $(uname -m)/ ; } || { echo ${ARHITECTURE}/ ; } )openshift-clients-[0-9]*.rpm
```

Reply:
>if you going to build image for arm64 on x86

I'd rather not to. This won't scale to other images and makes code a bit too bloated.